src/error.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::BTreeMap;
+use std::num::{ParseFloatError, ParseIntError};
+use std::result;
+use std::string::FromUtf8Error;
+
+use base64::DecodeError;
+use quick_xml::Error as XmlError;
+use thiserror::Error as ThisError;
+
+use crate::Value;
+
+/// Errors that can occur when trying to perform an XML-RPC request.
+///
+/// This can be a lower-level error (for example, the HTTP request failed), a problem with the
+/// server (maybe it's not implementing XML-RPC correctly), or just a failure to execute the
+/// operation.
+#[derive(ThisError, Debug)]
+pub enum Error {
+    /// The response could not be parsed. This can happen when the server doesn't correctly
+    /// implement the XML-RPC spec.
+    #[error("parse error: {0}")]
+    ParseError(#[from] ParseError),
```

Comment:
This was done on purpose to avoid exporting a ton of different types of errors. The public API only has what's available in the Error enum. The ParseError and EncodingError types should be private to the crate.